src/sys/windows/tcp.rs

```diff
@@ -149,6 +151,64 @@ pub(crate) fn set_linger(socket: TcpSocket, dur: Option<Duration>) -> io::Result
     }
 }
 
+pub(crate) fn set_keepalive(socket: TcpSocket, dur: Option<Duration>) -> io::Result<()> {
+    // Windows takes the keepalive timeout as a u32 of milliseconds.
+    let dur_ms = dur.map(|dur| {
+        let ms = dur.as_millis();
+        if ms > DWORD::max_value() as u128 {
+            winapi::um::winbase::INFINITE
```

Comment:
On Unix we use `i32::max_value()`, which I think we should here also (assuming `INFINITE` is actually infinite and not just the max value).

Reply:
`INFINITE` _is_ just the max value, but I thought it was better to use the winapi constants explicitly, just in case they are ever different?